misc/multiaddr/src/lib.rs

```diff
@@ -34,7 +34,7 @@ pub use errors::{Result, Error};
 pub use protocol::Protocol;
 
 /// Representation of a Multiaddr.
-#[derive(PartialEq, Eq, Clone, Hash)]
+#[derive(PartialEq, Eq, Clone, Default, Hash)]
```

Comment:
I don't think having an empty `Multiaddr` is a "reasonable" thing, so I don't think `Default` is a good idea.
However adding `Multiaddr::empty()` is fine.